### PR TITLE
Handle missing environment in GetCurrentDeployment

### PR DESCRIPTION
**Pull request by kiranmeduri**
#### Summary
Bugfix: Handle missing environment in GetCurrentDeployment

#### Testing

[x] Binary built locally and unit-tests pass (`make`)  
[x] Build in docker succeeds (`make release`)

New tests cover the changes: yes

#### Description for the changelog
Handle missing environment in GetCurrentDeployment

#### Licensing

This contribution is under the terms of the Apache 2.0 License: Yes

### CODE REVIEW

File: daemon-scheduler/pkg/deployment/environment_test.go [hunk 1]
```diff
@@ -185,6 +186,139 @@ func (suite *EnvironmentTestSuite) TestGetEnvironment() {
...
+	deployment.Status = types.DeploymentInProgress
+	environment.InProgressDeploymentID = deployment.ID
+	environment.Deployments[deployment.ID] = *deployment
+
+	suite.environmentStore.EXPECT().GetEnvironment(suite.ctx, environment.Name).
+		Return(environment, nil)
+
+	d, err := suite.environment.GetCurrentDeployment(suite.ctx, environment.Name)
+	assert.Nil(suite.T(), err, "Unexpected error when calling GetCurrentDeployment")
+	assert.Exactly(suite.T(), deployment.ID, d.ID, "Expected the deployment to match the expected one")
```
Comment (poojamaiya) [inline daemon-scheduler/pkg/deployment/environment_test.go]: match the expected one -> match the in progress deployment

File: daemon-scheduler/pkg/deployment/environment_test.go [hunk 2]
```diff
@@ -185,6 +186,139 @@ func (suite *EnvironmentTestSuite) TestGetEnvironment() {
...
+	deployment.Status = types.DeploymentPending
+	environment.PendingDeploymentID = deployment.ID
+	environment.Deployments[deployment.ID] = *deployment
+
+	suite.environmentStore.EXPECT().GetEnvironment(suite.ctx, environment.Name).
+		Return(environment, nil)
+
+	d, err := suite.environment.GetCurrentDeployment(suite.ctx, environment.Name)
+	assert.Nil(suite.T(), err, "Unexpected error when calling GetCurrentDeployment")
+	assert.Exactly(suite.T(), deployment.ID, d.ID, "Expected the deployment to match the expected one")
```
Comment (poojamaiya) [inline daemon-scheduler/pkg/deployment/environment_test.go]: match the expected one -> match the pending deployment

File: daemon-scheduler/pkg/deployment/environment_test.go [hunk 3]
```diff
@@ -185,6 +186,139 @@ func (suite *EnvironmentTestSuite) TestGetEnvironment() {
...
+}
+
+func (suite *EnvironmentTestSuite) TestGetCurrentDeploymentCompleted() {
+	environment, err := types.NewEnvironment("TestGetCurrentDeployment", "td-arn", "cluster-arn")
+	assert.Nil(suite.T(), err, "Unexpected error when creating environment")
+	deployment, err := types.NewDeployment("td-arn", uuid.NewRandom().String())
+	assert.Nil(suite.T(), err, "Unexpected error when creating deployment")
+	deployment.Status = types.DeploymentCompleted
+	environment.InProgressDeploymentID = deployment.ID
```
Comment (poojamaiya) [inline daemon-scheduler/pkg/deployment/environment_test.go]: So when the deployment completes, it doesn't clear the in progress deployment id?

File: daemon-scheduler/pkg/deployment/environment_test.go [hunk 4]
```diff
@@ -185,6 +186,139 @@ func (suite *EnvironmentTestSuite) TestGetEnvironment() {
...
+func (suite *EnvironmentTestSuite) TestDeleteEnvironment() {
+	environment, err := types.NewEnvironment("TestDeleteEnvironment", "td-arn", "cluster-arn")
+	assert.Nil(suite.T(), err, "Unexpected error when creating environment")
+	suite.environmentStore.EXPECT().GetEnvironment(suite.ctx, environment.Name).
+		Return(environment, nil)
+	suite.environmentStore.EXPECT().DeleteEnvironment(suite.ctx, *environment).
+		Return(nil).Times(1)
+
+	err = suite.environment.DeleteEnvironment(suite.ctx, environment.Name)
+	assert.Nil(suite.T(), err, "Unexpected error")
```
Comment (poojamaiya) [inline daemon-scheduler/pkg/deployment/environment_test.go]: Unexpected error -> Unexpected error deleting environment

File: daemon-scheduler/pkg/deployment/environment_test.go [hunk 5]
```diff
@@ -185,6 +186,139 @@ func (suite *EnvironmentTestSuite) TestGetEnvironment() {
...
+}
+
+func (suite *EnvironmentTestSuite) TestDeleteEnvironmentEmptyName() {
+	suite.environmentStore.EXPECT().GetEnvironment(suite.ctx, gomock.Any()).
+		Times(0)
+	suite.environmentStore.EXPECT().DeleteEnvironment(suite.ctx, gomock.Any()).
+		Times(0)
+
+	_, ok := suite.environment.DeleteEnvironment(suite.ctx, "").(types.BadRequestError)
+	assert.True(suite.T(), ok, "Expecting BadRequestError")
```
Comment (poojamaiya) [inline daemon-scheduler/pkg/deployment/environment_test.go]: Expecting BadRequestError -> Expecting BadRequestError when deleting environment with empty name

File: daemon-scheduler/pkg/deployment/environment_test.go [hunk 6]
```diff
@@ -185,6 +186,139 @@ func (suite *EnvironmentTestSuite) TestGetEnvironment() {
...
+	environment, err := types.NewEnvironment("TestDeleteEnvironmentGetEnvironmentReturnsEmpty", "td-arn", "cluster-arn")
+	assert.Nil(suite.T(), err, "Unexpected error when creating environment")
+
+	suite.environmentStore.EXPECT().GetEnvironment(suite.ctx, environment.Name).
+		Return(nil, nil)
+	suite.environmentStore.EXPECT().DeleteEnvironment(suite.ctx, *environment).
+		Times(0)
+
+	observedErr := suite.environment.DeleteEnvironment(suite.ctx, environment.Name)
+	assert.Nil(suite.T(), observedErr, "Unexpected error")
```
Comment (poojamaiya) [inline daemon-scheduler/pkg/deployment/environment_test.go]: I'm not sure I understand. If the environment requested to be deleted, it's not an error?

File: daemon-scheduler/pkg/deployment/environment_test.go [hunk 7]
```diff
@@ -185,6 +186,139 @@ func (suite *EnvironmentTestSuite) TestGetEnvironment() {
 	assert.Exactly(suite.T(), suite.environment1, env, "Expected the environment to match the expected one")
 }
 
+func (suite *EnvironmentTestSuite) TestGetCurrentDeploymentInProgress() {
+	environment, err := types.NewEnvironment("TestGetCurrentDeployment", "td-arn", "cluster-arn")
+	assert.Nil(suite.T(), err, "Unexpected error when creating environment")
```
Comment (kiranmeduri) [inline daemon-scheduler/pkg/deployment/environment_test.go]: done

File: daemon-scheduler/pkg/deployment/environment_test.go (hunk 3 above)
Comment (kiranmeduri) [inline daemon-scheduler/pkg/deployment/environment_test.go]: it is done by the monitor, but there can be a race. this makes it clear that even this case is handled

File: daemon-scheduler/pkg/deployment/environment_test.go (hunk 6 above)
Comment (kiranmeduri) [inline daemon-scheduler/pkg/deployment/environment_test.go]: that is how it is today (not sure if it is intentional).

File: daemon-scheduler/pkg/deployment/environment_test.go (hunk 5 above)
Comment (kiranmeduri) [inline daemon-scheduler/pkg/deployment/environment_test.go]: Done

**Comment by kiranmeduri** [comment]
I am closing this because my old repo is unlinked from blox/blox. I will figure out how to link it. until then use #69

**Pull request closed without merging**